scipy/sparse/compressed.py

```diff
@@ -513,13 +513,17 @@ def _mul_sparse_matrix(self, other):
 
         return self.__class__((data,indices,indptr),shape=(M,N))
 
-    def diagonal(self):
-        """Returns the main diagonal of the matrix
+    def diagonal(self, k=0):
+        """Returns the k-th diagonal of the matrix
```

Comment:
Instead of duplicating the docstring here, we should use the docstring from `spmatrix`. ([example](https://github.com/scipy/scipy/blob/9597f6a0b5d415d65f8ffa5ab30f32d7a3b74ca0/scipy/sparse/compressed.py#L115))